receiver/prometheusexecreceiver/end_to_end_test.go

```diff
@@ -0,0 +1,142 @@
+func TestEndToEnd(t *testing.T) {
+	// Load the config from the yaml file
+	factories, err := componenttest.ExampleComponents()
+	assert.NoError(t, err)
+
+	factory := &Factory{}
+	factories.Receivers[factory.Type()] = factory
+
+	config, err := configtest.LoadConfigFile(t, path.Join(".", "testdata", "config.yaml"), factories)
+
+	assert.NoError(t, err)
+	assert.NotNil(t, config)
+
+	// Receiver without exec key, error expected and checked within function
+	execErrorTest(t, config.Receivers["prometheus_exec"])
+
+	// Waitgroup to allow the goroutines to finish for the following end-to-end tests
+	var waitGroup sync.WaitGroup
+	waitGroup.Add(2)
+
+	// Normal test with port defined, expose metrics from fake exporter and make sure they're scraped/received
+	go endToEndScrapeTest(t, config.Receivers["prometheus_exec/end_to_end_test/1"], &waitGroup)
+
+	// Normal test with port undefined by user, same as previous test
+	go endToEndScrapeTest(t, config.Receivers["prometheus_exec/end_to_end_test/2"], &waitGroup)
+
+	waitGroup.Wait()
+}
+
+// execErrorTest makes sure the config passed throws an error, since it's missing the exec key
+func execErrorTest(t *testing.T, errorReceiverConfig configmodels.Receiver) {
+	wrapper := new(zap.NewNop(), errorReceiverConfig.(*Config), nil)
+
+	err := wrapper.Start(context.Background(), nil)
+	if err == nil {
+		t.Errorf("end_to_end_test.go didn't get error, was expecting one since this config has no 'exec' key")
+	}
+}
+
+// endToEndScrapeTest scrapes a test endpoint (test_prometheus_exporter.go) twice, and between each scrape yields the execution with Sleep() to wait for the subprocess (exporter) to restar
+// - wait time is about 1s - to fail and restart, meaning it verifies three things: the scrape is successful (twice), the process was restarted correctly when failed and the underlying
+// Prometheus receiver was correctly stopped and then restarted. For extra testing the metrics values are different every time the subprocess exporter is started
+// And the uniqueness of the metric scraped is verified
+func endToEndScrapeTest(t *testing.T, receiverConfig configmodels.Receiver, waitGroup *sync.WaitGroup) {
+	defer waitGroup.Done()
+
+	// Create the wrapper
+	sink := &exportertest.SinkMetricsExporterOld{}
+	wrapper := new(zap.NewNop(), receiverConfig.(*Config), sink)
+
+	// Initiate building the embedded configs and managing the subprocess with Start()
+	err := wrapper.Start(context.Background(), componenttest.NewNopHost())
+	if err != nil {
+		t.Errorf("end_to_end_test.go got error = %w", err)
+	}
+	defer wrapper.Shutdown(context.Background())
+
+	var metrics []consumerdata.MetricsData
+
+	// Make sure a first scrape works by checking for metrics in the test metrics exporter "sink", only return true when there are metrics
+	const waitFor = 15 * time.Second
+	const tick = 500 * time.Millisecond
```

Comment:
I've tested this multiple times, and since the code needs to spawn a subprocess that serves an endpoint, the Receiver needs to scrape that endpoint and push the metrics through the sink, it takes considerably more than 1s. Anyways, the `Eventually` will be exited as soon as the condition is met, it's just an upper limit for that given test, from what I understand! Also, given the longer time it takes to actually scrape, 10ms tick time is too fast and reduces performance considerably, since I guess it's constantly "context switching" to that goroutine and verifying the condition. The test just fails with 1s wait time, and it takes much longer with a 10ms tick time!